app/code/Magento/InventoryShipping/Plugin/Sales/ResourceModel/Order/Shipment/LoadSourceForShipmentPlugin.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryShipping\Plugin\Sales\ResourceModel\Order\Shipment;
+
+use Magento\Framework\Model\AbstractModel;
+use Magento\InventoryShipping\Model\ResourceModel\ShipmentSource\GetSourceCodeByShipmentId;
+use Magento\Sales\Model\ResourceModel\Order\Shipment as ShipmentResource;
+use Magento\Sales\Api\Data\ShipmentExtensionFactory;
+
+class LoadSourceForShipmentPlugin
+{
+    /**
+     * @var ShipmentExtensionFactory
+     */
+    private $shipmentExtensionFactory;
+
+    /**
+     * @var GetSourceCodeByShipmentId
+     */
+    private $getSourceCodeByShipmentId;
+
+    /**
+     * @param ShipmentExtensionFactory $shipmentExtensionFactory
+     * @param GetSourceCodeByShipmentId $getSourceCodeByShipmentId
+     */
+    public function __construct(
+        ShipmentExtensionFactory $shipmentExtensionFactory,
+        GetSourceCodeByShipmentId $getSourceCodeByShipmentId
+    ) {
+        $this->shipmentExtensionFactory = $shipmentExtensionFactory;
+        $this->getSourceCodeByShipmentId = $getSourceCodeByShipmentId;
+    }
+
+    /**
+     * @param ShipmentResource $subject
+     * @param ShipmentResource $result
+     * @param AbstractModel $shipment
+     * @return ShipmentResource
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterLoad(
+        ShipmentResource $subject,
+        ShipmentResource $result,
+        AbstractModel $shipment
+    ) {
+        $shipmentExtension = $shipment->getExtensionAttributes();
```

Comment:
it's very strange that Shipment implementation of get Extension attribute function looks like 
```php
    /**
     * {@inheritdoc}
     *
     * @return \Magento\Sales\Api\Data\ShipmentExtensionInterface|null
     */
    public function getExtensionAttributes()
    {
        return $this->_getExtensionAttributes();
    }
```
as it suppose to be implemented as 
```php
    /**
     * {@inheritdoc}
     *
     * @return \Magento\Catalog\Api\Data\ProductExtensionInterface
     */
    public function getExtensionAttributes()
    {
        $extensionAttributes = $this->_getExtensionAttributes();
        if (null === $extensionAttributes) {
            /** @var \Magento\Catalog\Api\Data\ProductExtensionInterface $extensionAttributes */
            $extensionAttributes = $this->extensionAttributesFactory->create(ProductInterface::class);
            $this->setExtensionAttributes($extensionAttributes);
        }
        return $extensionAttributes;
    }
```